cedar-policy-mcp-schema-generator/src/description/err.rs

```diff
@@ -0,0 +1,211 @@
+    ParseError(#[from] super::parser::err::ParseError),
+
+    /// Deserializer encountered an unexpected JSON type
+    #[error("Encountered unexpected JSON type while deserializing {}.", .0.content_type)]
+    #[diagnostic(transparent)]
+    UnexpectedType(LocationFound),
+
+    /// Deserializer did not find an expected attribute
+    #[error(transparent)]
+    #[diagnostic(transparent)]
+    MissingExpectedAttribute(MissingExpectedAttributeError),
+
+    /// Deserializer found an unexpected value
+    #[error("Unexpected value.")]
+    #[diagnostic(transparent)]
+    UnexpectedValue(LocationFound),
+
+    /// Deserializer could not open the provided JSON file
+    #[error("Error reading {}: {}", .0.file_name.display(), .0.error)]
+    #[diagnostic()]
+    ReadError(ReadError),
+}
+
+impl DeserializationError {
+    /// Construct a new `DeserizlizerError` signaling an Unexpected JSON type was encountered while deserializing
+    pub(crate) fn unexpected_type(
+        json_value: &LocatedValue,
+        msg: &str,
+        content_type: ContentType,
+    ) -> Self {
+        let loc = json_value.as_loc().clone();
+        Self::UnexpectedType(LocationFound {
+            src: loc,
+            label: "Found".to_string(),
+            msg: msg.to_string(),
+            code: "deserialization::unexpected_type".to_string(),
+            content_type,
+        })
+    }
+
+    /// Construct a new `DeserizlizerError` signaling an expected attribute is missing from a JSON Object
+    pub(crate) fn missing_attribute(
+        json_value: &LocatedValue,
+        expected_key: &str,
+        aliases: Vec<String>,
+    ) -> Self {
+        let loc = json_value.as_loc().clone();
+        let existing_keys = json_value.get_object().map_or(Vec::new(), |obj| {
+            obj.keys().map(|key| key.as_loc().into()).collect()
+        });
+        Self::MissingExpectedAttribute(MissingExpectedAttributeError {
+            loc,
+            expected_key: expected_key.to_string(),
+            aliases,
+            existing_keys,
+        })
+    }
+
+    /// Construct a new `DeserizlizerError` signaling an Unexpected JSON value was encountered while deserializing
+    pub(crate) fn unexpected_value(
+        json_value: &LocatedValue,
+        msg: &str,
+        content_type: ContentType,
+    ) -> Self {
+        let loc = json_value.as_loc().clone();
+        Self::UnexpectedType(LocationFound {
+            src: loc,
+            label: "Found".to_string(),
+            msg: msg.to_string(),
+            code: "deserialization::unexpected_value".to_string(),
+            content_type,
+        })
+    }
+
+    /// Construct a new `DeserizlizerError` signaling that input JSON file could not be read
+    pub(crate) fn read_error(file_name: PathBuf, error: String) -> Self {
+        Self::ReadError(ReadError { file_name, error })
+    }
+}
+
+#[derive(Error, Debug)]
+#[error("Missing expected attribute")]
+pub struct MissingExpectedAttributeError {
```

Comment:
Can we please call it
```suggestion
pub struct MissingAttributeError {
```
?

"expected" is kind of redundant: the attribute was missing because you expected it.

Reply:
(this will allow for errors to be less similar to each other)